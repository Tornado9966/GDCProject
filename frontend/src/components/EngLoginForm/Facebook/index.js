import { connect } from 'react-redux';

import { userActions } from 'redux/actions';
import Facebook from './Facebook';

function mapState(state = {}) {
    const { loggedIn, email, firstName, userImage } = state.auth;
    return {
        loggedIn, email, firstName, userImage
    };
}

const mapDispatchToProps = {
    socialFbLogin: userActions.socialFbLogin
};

export default connect(mapState, mapDispatchToProps)(Facebook);