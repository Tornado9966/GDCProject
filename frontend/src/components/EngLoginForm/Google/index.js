import { connect } from 'react-redux';

import { userActions } from 'redux/actions';
import Google from './Google';

function mapState(state = {}) {
    const { loggedIn, email, firstName, userImage } = state.auth;
    return {
        loggedIn, email, firstName, userImage
    };
}   

const mapDispatchToProps = {
    socialGLogin: userActions.socialGLogin
};

export default connect(mapState, mapDispatchToProps)(Google);