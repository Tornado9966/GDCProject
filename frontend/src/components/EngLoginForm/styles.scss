@import "assets/_colors.scss";

.ui {
  .dimmer {
    background-color: rgba(255,255,255,.85);
  }
  &.login-wrap {
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    height: 80%;
    padding: 30px;

    .login-container {
      max-width: 400px;
    }
    .login-header {
      color: $yellow;
      text-align: center;
      font-size: 24px;
    }
    .login-form {
      font-size: 16px;
    }
  }
  &.login-success {
    background-color: $white;
    font-size: 16px;
    width: 100px;
    color: $yellow;
    border: 2px solid $yellow;

    &:hover {
      background-color: $yellow;
      color: #fff;
    }
    &:active {
      background-color: $dark_yellow;
      color: #fff;
      border: 2px solid $dark_yellow;
    }
  }

  &.login-button {
    background-color: $white;
    font-size: 16px;
    width: 60% !important;
    color: $yellow;
    border: 2px solid $yellow;

    &:hover {
      background-color: $yellow;
      color: #fff;
    }
    &:active {
      background-color: $dark_yellow;
      color: #fff;
      border: 2px solid $dark_yellow;
    }
  }
}
.socials-wrap {
  display: flex;
  justify-content: center;
}

.socials-google {
  margin: 0 0 0 5px;
}
.socials-facebook {
  margin: 0 5px 0 0;
  width: 60%;
  overflow: hidden;
  border-radius: 4px;
}

div.socials-facebook > div > span > button {
  padding: 4px 8px;
  font-size: 14px;
}
.kep-login-facebook {
  width: 100%;
  line-height: 17px;
}
.socials-google > div > button{
  height: 44px;
}
.socials-google > div > button > div{
  padding: 7px !important;
}

.socials-google > div > button > span {
  padding: 8px 10px 3px 0px !important;
}
.login-password {
  padding: 0 0 1em 0;

  .forgot-password {
    &:hover {
      cursor: pointer;
    }
  }
}
.ui.stacked.segment { 
  &:before, &:after {
      display: none;
  }
}