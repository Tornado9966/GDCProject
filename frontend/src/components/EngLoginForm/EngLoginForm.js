import React from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { some as _some } from 'lodash';
import {
  Button,
  Form,
  Grid,
  Header,
  Message,
  Segment,
  List
} from 'semantic-ui-react';

import { isInvalid } from 'components/utils/isInvalid';
import NotificationMessage from 'components/utils/NotificationMessage';
import * as constText from 'constants/text-constants';
import Facebook from './Facebook';
import Google from './Google';

import './styles.scss';

export class EngLoginForm extends React.Component {
  static propTypes = {
    loggedIn: PropTypes.bool,
    loading: PropTypes.bool,
    showError: PropTypes.bool
  };

  static defaultProps = {
    loggedIn: false,
    loading: false,
    showError: false
  };

  state = {
    email: '',
    password: '',
    forgotPassword: false,
    notValid: {
      email: true,
      password: true
    }
  };

  disableSubmitBtn = () => {
    return this.state.forgotPassword
      ? this.state.notValid.email
      : _some(this.state.notValid, item => item !== false);
  };

  toggleForm = () => {
    this.setState(prevState => ({
      ...prevState,
      forgotPassword: !prevState.forgotPassword,
      emailSent: false,
      sendEmailError: false
    }));
  };

  handleChange = (event, { name, value }) => {
    const sendEmailError = Boolean(isInvalid(value, 'login'));

    this.setState(prevState => ({
      [name]: value,
      notValid: {
        ...prevState.notValid,
        [name]: sendEmailError
      }
    }));
  };

  handleSubmit = () => {
    const { email, password } = this.state;

    this.state.forgotPassword
      ? this.props.sendEmail(email)
      : this.props.login(email, password);
  };

  render() {
    const { forgotPassword } = this.state;
    const {
      loggedIn,
      loading,
      successMessage,
      showError
    } = this.props;

    return (
      loggedIn
        ? (<Redirect to='/' />)
        : (
          <Grid className="login-wrap">
            <Grid.Column className="login-container">
              <Header className="login-header">
                {forgotPassword ? constText.resetPassword : constText.headerText}
              </Header>

              {
                showError &&
                (
                  <NotificationMessage error={Boolean(true)} errorText={constText.loginErrorText} />
                )
              }

              {
                successMessage &&
                (
                  <Message positive>
                    <Message.Header>{constText.confirmEmail}</Message.Header>
                  </Message>
                )
              }

              <Form
                loading={loading}
                onSubmit={this.handleSubmit}
                className="login-form"
              >
                <Segment stacked>
                  {
                    forgotPassword &&
                    <List className='login-password'>
                      <List.Item>
                        {constText.enterEmailToResetPass}
                      </List.Item>
                    </List>
                  }
                  <Form.Input
                    fluid
                    name='email'
                    type='email'
                    icon='user'
                    iconPosition='left'
                    placeholder='E-mail address'
                    onChange={this.handleChange}
                  />
                  {
                    forgotPassword
                      ? <List>
                        <List.Item
                          as='a'
                          className='forgot-password'
                          onClick={this.toggleForm}
                        >
                          {constText.returnToLogin}
                        </List.Item>
                        </List>
                      : <div className='login-password'>
                        <Form.Input
                          fluid
                          name='password'
                          type='password'
                          icon='lock'
                          iconPosition='left'
                          placeholder='Password'
                          onChange={this.handleChange}
                        />
                        <List.Item
                          as='a'
                          className='forgot-password'
                          onClick={this.toggleForm}
                        >
                          {constText.forgotPassword}
                        </List.Item>
                        </div>
                  }
                  <Button
                    className="login-button"
                    disabled={this.disableSubmitBtn()}
                  >
                    {
                      forgotPassword
                        ? constText.sendEmailToResetPass
                        : constText.loginBtnText
                    }
                  </Button>
                </Segment>
              </Form>
              <Message>
                {constText.signUpText}
                <a href={constText.linkToRegistration}>
                  {constText.signUpLinkText}
                </a>
              </Message>
              <Header as='h5'>
                <div className="socials-wrap">
                  <div className="socials-facebook"><Facebook /></div>
                  <div className="socials-google"><Google /></div>
                </div>
              </Header>
            </Grid.Column>
          </Grid>
        )
    );
  }
}