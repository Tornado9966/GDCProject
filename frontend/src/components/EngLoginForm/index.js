import {connect} from 'react-redux';

import { userActions } from 'redux/actions';
import { EngLoginForm } from './EngLoginForm';

function mapState({auth: {loggedIn}, loading, successMessage, showError}) {
  return {
    loggedIn,
    loading,
    successMessage,
    showError
  };
}

const mapDispatchToProps = {
  login: userActions.login,
  logout: userActions.logout,
  sendEmail: userActions.sendEmail
};
  
export default connect(mapState, mapDispatchToProps)(EngLoginForm);
