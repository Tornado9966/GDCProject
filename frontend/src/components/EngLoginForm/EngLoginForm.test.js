import React from 'react';
import { shallow } from 'enzyme';

import { EngLoginForm } from './EngLoginForm';

const setUp = (props = {}) => {
    const component = shallow(<EngLoginForm {...props} />);
    return component;
};

describe('EngLoginForm', () => {
    let component;
    const props = {
      loggedIn: false,
      loading: false,
      showError: false
    };

    beforeEach(() => {
        component = setUp();
    });

    it('matches the snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('should render correctly in "debug" mode', () => {
        const component = shallow(<EngLoginForm debug />);
        expect(component).toMatchSnapshot();
    });
    
    it('should render correctly with no props', () => {
        const component = shallow(<EngLoginForm />);
        
        expect(component).toMatchSnapshot();
    });

    it('should render correctly with boolean prop', () => {
        const component = shallow(<EngLoginForm {...props} />);
        expect(component).toMatchSnapshot();
    });

    it('it should contain login-wrap element', () => {
        const wrapper = component.find('.login-wrap');
        expect(wrapper.length).toBe(1);
    });
});