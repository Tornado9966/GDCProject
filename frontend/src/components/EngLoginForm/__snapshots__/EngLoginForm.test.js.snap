// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EngLoginForm matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`EngLoginForm should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`EngLoginForm should render correctly with boolean prop 1`] = `ShallowWrapper {}`;

exports[`EngLoginForm should render correctly with no props 1`] = `ShallowWrapper {}`;

exports[`EngUserDropdown matches the snapshot 1`] = `ShallowWrapper {}`;
